pkg/build/config.go

```diff
@@ -10,14 +10,27 @@ import (
 	"github.com/openshift/source-to-image/pkg/scm/git"
 )
 
-// GenerateConfigFromLabels generates the S2I Config struct from the Docker
+// GenerateConfigFromBuilderImageLabels generates the S@I Config struct from the Docker
 // image labels.
-func GenerateConfigFromLabels(config *api.Config, metadata *docker.PullResult) error {
+func GenerateConfigFromBuilderImageLabels(config *api.Config, metadata *docker.PullResult) error {
 	if config == nil {
-		return errors.New("config must be provided to GenerateConfigFromLabels")
+		return errors.New("config must be provided to GenerateConfigFromBuilderImageLabels")
 	}
 	if metadata == nil {
-		return errors.New("image metadata must be provided to GenerateConfigFromLabels")
+		return errors.New("image metadata must be provided to GenerateConfigFromBuilderImageLabels")
+	}
+
+	return nil
+}
+
+// GenerateConfigFromApplicationImageLabels generates the S2I Config struct from the Docker
+// image labels.
+func GenerateConfigFromApplicationImageLabels(config *api.Config, metadata *docker.PullResult) error {
```

Comment:
What is the difference between a builder image and application image in this context ?

Reply:
There are different expectations for labels in the builder image and application image. In that particular case, Application Image's labels are reused in `rebuild`, where `build` should honour Builder Image's labels.